### PR TITLE
Added HeadlessPluginContext, Canvas3DRenderer, image-renderer example

### PR DESCRIPTION
**Pull request by midlik**
# Description

I created `HeadlessPluginContext`, which allows running Mol* Plugin outside a browser (in Node.js).
An example of its usage is shown in `src/examples/image-renderer/`.

## Actions

- [x] Added description of changes to the `[Unreleased]` section of `CHANGELOG.md`
- [x] Updated headers of modified files
- [x] Added my name to `package.json`'s `contributors`

### CODE REVIEW

**Comment by russellp17** [comment]
this is cool! I'd expect these sorts of dependencies to be under [optionalDependencies](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#optionaldependencies) in package.json instead of unlisted, so that library users can decide whether to skip them if necessary - is there a reason not to do that?

**Comment by dsehnal** [comment]
That's a good point @russellp17, listing these in optional dependencies makes sense. @midlik  can you please update this?

**Comment by midlik** [comment]
Thank you for your suggestion, @russellp17, I added them to optionalDependencies.

**Review by dsehnal — APPROVED** [review]

File: src/mol-util/nodejs-browser-io.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2018-2020 mol* contributors, licensed under MIT, See LICENSE file for more info.
```
Comment (arose) [inline src/mol-util/nodejs-browser-io.ts:2]: date...

File: src/mol-util/nodejs-browser-io.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ *
+ * @author Adam Midlik <midlik@gmail.com>
+ *
+ * Implements some browser-only global variables for Node.js environment.
+ * These workarounds will also work in browsers as usual.
+ */
+
+
+/** Determines whether the current code is running in Node.js or in a browser */
+export const RUNNING_IN_NODEJS = typeof document === 'undefined';
```
Comment (arose) [inline src/mol-util/nodejs-browser-io.ts:12]: there other runtime times... this checks if it is running in a browser, to check for node do

```
const isNode =
  typeof process !== "undefined" &&
  process.versions != null &&
  process.versions.node != null;
```

File: src/mol-util/lazy-imports.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2019-2022 mol* contributors, licensed under MIT, See LICENSE file for more info.
```
Comment (arose) [inline src/mol-util/lazy-imports.ts:2]: date...

File: src/mol-util/data-source.ts [hunk 4]
```diff
@@ -7,21 +7,14 @@
  * Adapted from LiteMol
  */
 
+import * as fs from 'fs';
```
Comment (arose) [inline src/mol-util/data-source.ts:10]: Importing a node module like this causes issues in the browser and forces you to handle it with a bundler like webpack. Please find another way.

File: src/mol-canvas3d/renderer.ts [hunk 5]
```diff
@@ -0,0 +1,223 @@
+/**
+ * Copyright (c) 2019-2022 mol* contributors, licensed under MIT, See LICENSE file for more info.
```
Comment (arose) [inline src/mol-canvas3d/renderer.ts:2]: date

File: src/mol-canvas3d/renderer.ts [hunk 6]
```diff
@@ -0,0 +1,223 @@
+/**
```
Comment (arose) [inline src/mol-canvas3d/renderer.ts:1]: I would move this to mol-plugin/util/headless-screenshot.ts

File: src/mol-canvas3d/renderer.ts [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+
+
+const lazyImports = LazyImports.create('gl', 'jpeg-js', 'pngjs') as {
+    'gl': typeof import('gl'),
+    'jpeg-js': typeof import('jpeg-js'),
+    'pngjs': typeof import('pngjs'),
+};
+
+
+export type ImageRendererOptions = {
```
Comment (arose) [inline src/mol-canvas3d/renderer.ts:37]: should have same name as the class it is used with

File: src/mol-canvas3d/renderer.ts [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+
+export type RawImageData = {
+    data: Uint8ClampedArray,
+    width: number,
+    height: number,
+}
+
+
+/** To render Canvas3D when running in Node.js (without DOM) */
+export class Canvas3DRenderer {
```
Comment (arose) [inline src/mol-canvas3d/renderer.ts:51]: I would call this HeadlessScreenshotHelper

File: src/mol-util/nodejs-browser-io.ts [hunk 9]
```diff
@@ -0,0 +1,60 @@
+/**
```
Comment (arose) [inline src/mol-util/nodejs-browser-io.ts:1]: I would call this file `nodesjs-shims.ts`, more generic

**Review by arose — COMMENTED** [review]
A bit late, but here is my review. Nice addition, thanks @midlik.

**Comment by midlik** [comment]
Thank you @arose , I will address your feedback in another PR.

File: src/mol-util/data-source.ts (hunk 4 above)
Comment (midlik) [reply src/mol-util/data-source.ts:10]: OK, now importing 'fs' and 'path' via LazyImports

**Pull request merged**